DifferentialDrive.md

```diff
@@ -0,0 +1,545 @@
+In Faults and Total Faults, we can see that we have indeed ran our checks and since there are no warnings, it is safe to assume that in this case the checks have passed. 
+- To see this in Elastic, click `Test` and then head to Elastic. Right click, click add widgets, click `Faults` and drag and drop `Total Faults` and `Active Faults` onto the layout. You should see the following if the tests have passed. 
+
+![](image-2.png)
+
+Same with Unit Tests, it's good to have system checks for all runnable parts of the robot to ensure the robot is fully ready for the game. As a notice, please make sure to run system checks while the robot is ON the robot cart. 
+### Simulation and Logging
+To finish our project up, we are going to simlute the drive we've made. Please read the [Simulation reference sheet](https://github.com/SciBorgs/SciGuides/blob/main/Simulation.md) before continuing on with this. 
+
+Setting Up the Simulation
+To simulate the drivetrain, we’re going to use the DifferentialDrivetrainSim class. This simulation will model the physical characteristics of our robot—like the motors, mass, and wheel dimensions—so we can see how the code will affect the robot in a virtual environment.
+
+Here’s how we set it up in our constructor:
+```java
+private final DifferentialDrivetrainSim driveSim;
+...
+driveSim =
+    new DifferentialDrivetrainSim(
+        DCMotor.getMiniCIM(2),
+        GEARING,
+        MOI,
+        DRIVE_MASS.in(Kilograms),
+        WHEEL_RADIUS,
+        TRACK_WIDTH,
+        STD_DEVS);
+```
+- Remember that all of these values would be stored in `driveConstants`:
+  ```
+  public static final double TRACK_WIDTH = 0.7112;
+  public static final double WHEEL_RADIUS = Units.inchesToMeters(3);
+  public static final double GEARING =  8;
+  public static final double MOI = 7.5;
+  public static final Measure<Mass> DRIVE_MASS = Kilograms.of(60.0);
+  public static final Matrix<N7, N1> STD_DEVS = VecBuilder.fill(0, 0, 0, 0, 0, 0, 0);
+  ```
+By setting up the simulation this way, we’re creating a virtual model of our robot that behaves similarly to how it would in real life.
+
+Before we start periodically updating our sim values, make sure to set the voltage to our sim drive in the `setSpeed` method: 
+```java
+driveSim.setInputs(leftVoltage, rightVoltage);
+```
+
+#### Simulation in Action: `simulationPeriodic`
+The `simulationPeriodic` method is where we update the simulation. This method is called regularly during the robot’s operation to keep the simulated sensors and drivetrain in sync with the rest of the code. We must put a `@Override` annotation above the method as it is inherited from extending to `subsystemBase`.
+```java
+@Override
+public void simulationPeriodic(){
+  driveSim.update(0.02);  // Update the drivetrain simulation every 20ms
+  leftEncoder.setPosition(driveSim.getLeftPositionMeters());
+  rightEncoder.setPosition(driveSim.getRightPositionMeters());
+  gyroSim.setAngle(-driveSim.getHeading().getDegrees());
```

Comment:
gyroSim isn't created anywhere, nor should it be. You can probably just make a double or Rotation2d that keeps track.

Reply:
Also just making sure: you have this project somewhere in your folders and have tested that everything works, right? I was thinking you might want to add some gifs of the sim in the beginning to engage people from the beginning